compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -444,7 +444,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     ) -> InterpResult<'tcx, (MPlaceTy<'tcx, M::PointerTag>, u64)> {
         // Basically we just transmute this place into an array following simd_size_and_type.
         // This only works in memory, but repr(simd) types should never be immediates anyway.
-        assert!(base.layout.ty.is_simd());
+        assert!(
```

Comment:
I think if we change anything, we should change this assert into an `if` and a `throw_ub_format`. I'm not sure we should change anything though. @workingjubilee is it ever expected that we'll allow calling SIMD intrinsics from stable code?

Reply:
Note that codegen accepts this call silently and likely with UB. If we do any fix for intrinsics, we should probably look into more general solutions, but previously we've decided to just keep ICEing on intrinsic mis-use as it's an infinite rabbit hole otherwise.

Reply:
Just double checking I'm understanding right, you're saying that because `extern "platform-intrinsics"` is likely going to be permanently unstable, an error like this would only be hit by another bug in the compiler, or by nightly code, so the difference between an ICE and a proper error message isn't as significant as for stable code?
But because swapping out an `assert` for `throw_ub_format` is a super easy change, it's still marginally better? Or am I misunderstanding?

Reply:
No misunderstanding, that's exactly what I meant. Of course this is just parrotting old information with some sprinkles of convenience on top. So I'm perfectly fine reevaluating this if you feel otherwise. We should also let @RalfJung  chime in before doing anything, as this may get into a game of whack-a-mole if we keep removing such ICEs.

Reply:
I don't have strong objections to replacing this `assert` by a `throw` -- it doesn't make the code harder to read or maintain. This should not give rise to the expectation that all ICEs due to misused intrinsics are bugs though, I don't want to see the codebase littered with such checks.

throw_ub seems wrong though; an "invalid program" error seems more accurate. We don't have throw_invalid_format yet, but that should not be too hard to add.

Reply:
> I think if we change anything, we should change this assert into an `if` and a `throw_ub_format`. I'm not sure we should change anything though. @workingjubilee is it ever expected that we'll allow calling SIMD intrinsics from stable code?

No, it is never intended that users will touch these, and we are at least somewhat considering if we can successfully deprecate `#[repr(simd)]` as a concept entirely (as it is also unstable) and just have it be the suite of stable `core::simd` types.

Reply:
Oh and if we change this, `mplace_to_simd` should probably throw the same error.

(operand_to_simd still needs its own check as otherwise assert_mem_place could ICE.)